cluster-autoscaler/cloudprovider/cloudifyprovider/scale_provider.go

```diff
@@ -57,69 +117,126 @@ func (clsp *CloudifyScaleProvider) Refresh() error {
-			// check runtime properties
-			if cloudInstance.GetStringProperty("hostname") != kubeNode.Name {
-				continue
+	if items == nil {
+		glog.Errorf("no deployments found !!!")
+	}
+	for _, dep := range items {
+		if dep["deployment_type"] == "node" {
+			glog.V(4).Infof("NodeGroupForNode(%v.%v)", dep["id"], kubeNode.Name)
+			groupedInstances, err := clsp.client.GetDeploymentInstancesScaleGrouped(
+				dep["id"], dep["node_data_type"])
+
+			if err != nil {
+				glog.Errorf("Cloudify error: %s\n", err.Error())
+				return nil, err
 			}
 
-			return NodeToNodeGroup(clsp.client, clsp.deploymentID, groupName), nil
+			for groupName, cloudInstances := range groupedInstances {
+				// search instance
+				for _, cloudInstance := range cloudInstances.Items {
+					// check runtime properties
+					if cloudInstance.GetStringProperty("hostname") != kubeNode.Name {
+						continue
+					}
+
+					return NodeToNodeGroup(clsp.client, dep["id"], groupName, dep["node_data_type"]), nil
+				}
+			}
+			glog.Warningf("NodeGroupForNode(%v.%v): Skiped, check list of members in scale group", dep["id"], kubeNode.Name)
 		}
 	}
-	glog.Warningf("NodeGroupForNode(%v.%v): Skiped, check list of members in scale group", clsp.deploymentID, kubeNode.Name)
+
 	return nil, nil
 }
 
 // Pricing returns pricing model for this cloud provider or error if not available.
 func (clsp *CloudifyScaleProvider) Pricing() (cloudprovider.PricingModel, errors.AutoscalerError) {
-	glog.Errorf("?Pricing: %v", clsp.deploymentID)
+	items, err := clsp.GetDeploymentNodeInfo()
+	if err != nil {
+		glog.Errorf("Cloudify error: %s\n", err.Error())
+	}
+
+	if items == nil {
+		glog.Errorf("no deployments found !!!")
+	}
+
+	for _, dep := range items {
+		glog.Errorf("?Pricing: %v", dep["id"])
+
+	}
 	return nil, cloudprovider.ErrNotImplemented
 }
 
 // GetAvailableMachineTypes get all machine types that can be requested from the cloud provider.
 // Implementation optional.
 func (clsp *CloudifyScaleProvider) GetAvailableMachineTypes() ([]string, error) {
-	glog.Errorf("?GetAvailableMachineTypes: %v", clsp.deploymentID)
+	items, err := clsp.GetDeploymentNodeInfo()
+	if err != nil {
+		glog.Errorf("Cloudify error: %s\n", err.Error())
+	}
+
+	if items == nil {
+		glog.Errorf("no deployments found !!!")
+	}
+
+	for _, dep := range items {
+		glog.Errorf("?GetAvailableMachineTypes: %v", dep["id"])
+
+	}
 	return []string{}, cloudprovider.ErrNotImplemented
 }
 
 // NewNodeGroup builds a theoretical node group based on the node definition provided. The node group is not automatically
 // created on the cloud provider side. The node group is not returned by NodeGroups() until it is created.
 func (clsp *CloudifyScaleProvider) NewNodeGroup(machineType string, labels map[string]string, extraResources map[string]resource.Quantity) (cloudprovider.NodeGroup, error) {
-	glog.Errorf("?NewNodeGroup(%v): %+v %+v %+v", clsp.deploymentID, machineType, labels, extraResources)
```

Comment:
Looks as debug code, could we drop it?

Reply:
We don't need it since I change how we handle deployment and no need to have it here
@0lvin

Reply:
@mabuaisha So remove :-)